pytensor/tensor/rewriting/subtensor.py

```diff
@@ -648,7 +648,7 @@ def local_subtensor_of_alloc(fgraph, node):
     # Slices to take from val
     val_slices = []
 
-    for i, (sl, dim) in enumerate(zip(slices, dims, strict=False)):
+    for i, (sl, dim) in enumerate(zip(slices, dims[: len(slices)], strict=True)):
```

Comment:
Like my previous comment, I find this less readable. The `strict=False` indicates clearly that we don't expect the sequences to necessarily have the same length?

Reply:
Because they are ordered correctly yes, and presumably what comes after doesn't matter. It's quite common in Subtensor operations / rewrites

Reply:
I don't mind reverting this, but just to argue a bit in the favor of `strict=True`, I think an additional advantage of this approach is that it makes it clearer which one of the two lists is supposed to be shorter.